x-pack/plugins/apm/server/routes/agent_keys/route.ts

```diff
@@ -74,10 +76,40 @@ const invalidateAgentKeyRoute = createApmServerRoute({
   },
 });
 
+const createAgentKeyRoute = createApmServerRoute({
+  endpoint: 'POST /apm/agent_keys',
+  options: { tags: ['access:apm', 'access:apm_write'] },
+  params: t.type({
+    body: t.intersection([
+      t.partial({
+        sourcemap: toBooleanRt,
+        event: toBooleanRt,
+        agentConfig: toBooleanRt,
+      }),
```

Comment:
As mentioned in the docs, I think this should simply take a list of the privileges, restricted to the 3 APM privileges.